scenes/game_state/game_state.gd

```diff
@@ -3,5 +3,42 @@
 @tool
 extends Node
 
+## Emited when a new item is collected, even if it wasn't added to the
+## inventory due to it being already there.
+signal item_collected(item: InventoryItem)
+
+## Global inventory, used to track the items the player obtains and that
+## can be added to the loom.
 @export var inventory: Inventory = Inventory.new()
 @export var current_spawn_point: NodePath
+
+## Quest's items. Used to track the progress withing a story quest.
+var story_quest_inventory: Inventory = Inventory.new()
+
+
+## Resets the [member story_quest_inventory]. This needs to be called when
+## a new story quest starts.
+func quest_started() -> void:
+	story_quest_inventory.clear()
+
+
+## Adds the [InventoryItem] to the [member inventory] and the
+## [member story_quest_inventory], provided it wasn't already there.
+func item_was_collected(item: InventoryItem) -> void:
+	if not inventory.has_item(item):
+		inventory.add_item(item)
+	if not story_quest_inventory.has_item(item):
+		story_quest_inventory.add_item(item)
```

Comment:
It looks like you are using inventory as a set primitive. Since GDScript has none https://github.com/godotengine/godot-proposals/issues/867 why not move that logic to the Inventory resource? Then from here you just add the item, and Inventory does the job.

Reply:
Yeah, I wasn't completely sure if we'd like to support being able to collect items again or not 🤔. I will move it to the Inventory!, and we can change that behavior later if we find out we want to support repeated items.